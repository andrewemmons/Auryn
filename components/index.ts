/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

export { BackButton } from './backButton';
export { BackgroundImage } from './backgroundImage';
export { DiscoverContainer } from './discoverContainer';
export { List } from './list';
export { ListItem } from './listitem';
export { Timeline } from './timeline';
export { ToggleButton } from './toggleButton';
export { ToggleGroup } from './toggleGroup';
export { ToggleGroupList } from './toggleGroupList';
export { TvContainer } from './tvContainer';
export { Error } from './error';
