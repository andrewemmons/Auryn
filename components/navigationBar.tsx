/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { ToggleButton } from '.';
import { ToggleButtonProps } from './toggleButton';
import { Composition, ScrollRef } from '@youi/react-native-youi';
import { View, StyleSheet, ScrollViewProps } from 'react-native';

interface NavigationBarProps extends ScrollViewProps {
  initialToggleIndex: number;
  onPressItem: (index: number) => void;
  name: string;
  focusable?: boolean;
};

export class NavigationBar extends React.PureComponent<NavigationBarProps, { activeButtonIndex: number }> {
  state = { activeButtonIndex: -1 };

  initialToggleIndex = this.props.initialToggleIndex;

  render() {
    // eslint-disable-next-line consistent-return
    const data = React.Children.map(this.props.children, (child, index) => {
      const typedChild = child as React.ReactElement<ToggleButtonProps>;
      if (typedChild.type === ToggleButton) {
        const button = React.cloneElement(typedChild, {
          onPress: () => {
            this.initialToggleIndex = -1;
            if (this.state.activeButtonIndex === index) return;

            this.setState({ activeButtonIndex: index });

            if (this.props.onPressItem)
              this.props.onPressItem(index);
             else if (typedChild.props.onPress)
             typedChild.props.onPress(index);
          },
          toggled: this.initialToggleIndex === index || this.state.activeButtonIndex === index,
          focusOnMount: this.initialToggleIndex === index,
          focusable: this.props.focusable,
          name: 'Btn-Nav-List',
        });
        return <View><Composition source="Auryn_Btn-Nav-List-Container">{button}</Composition></View>;
      };
    });

    return <ScrollRef {...this.props}>
        <View style={styles.buttonContainer}>
          {data}
        </View>
      </ScrollRef>;
  }
}

const styles = StyleSheet.create({
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
});
