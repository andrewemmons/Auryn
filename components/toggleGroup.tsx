/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { ToggleButton } from '.';
import { ToggleButtonProps } from './toggleButton';

interface ToggleGroupProps {
  initialToggleIndex: number;
  onPressItem: (index: number) => void;
};

export class ToggleGroup extends React.PureComponent<ToggleGroupProps, { activeButtonIndex: number }> {
  state = { activeButtonIndex: -1 };

  initialToggleIndex = this.props.initialToggleIndex;

  render() {
    // eslint-disable-next-line consistent-return
    return React.Children.map(this.props.children, (child, index) => {
      const typedChild = child as React.ReactElement<ToggleButtonProps>;
      if (typedChild.type === ToggleButton) {
        return React.cloneElement(typedChild, {
          onPress: () => {
            this.initialToggleIndex = -1;
            if (this.state.activeButtonIndex === index) return;

            this.setState({ activeButtonIndex: index });

            if (this.props.onPressItem)
              this.props.onPressItem(index);
             else if (typedChild.props.onPress)
             typedChild.props.onPress(index);
          },
          toggled: this.initialToggleIndex === index || this.state.activeButtonIndex === index,
          focusOnMount: this.initialToggleIndex === index,
        });
      };
    });
  }
}
