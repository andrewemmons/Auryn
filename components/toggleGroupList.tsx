/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { ToggleButton } from '.';
import { ToggleButtonProps } from './toggleButton';
import { Composition, ScrollRef } from '@youi/react-native-youi';
import { View, StyleSheet } from 'react-native';

interface ToggleGroupListProps {
  initialToggleIndex: number;
  onPressItem: (index: number) => void;
  name: string;
};

export class ToggleGroupList extends React.PureComponent<ToggleGroupListProps, { activeButtonIndex: number }> {
  state = { activeButtonIndex: -1 };

  initialToggleIndex = this.props.initialToggleIndex;

  render() {
    // eslint-disable-next-line consistent-return
    const data = React.Children.map(this.props.children, (child, index) => {
      const typedChild = child as React.ReactElement<ToggleButtonProps>;
      if (typedChild.type === ToggleButton) {
        const button = React.cloneElement(typedChild, {
          onPress: () => {
            this.initialToggleIndex = -1;
            if (this.state.activeButtonIndex === index) return;

            this.setState({ activeButtonIndex: index });

            if (this.props.onPressItem)
              this.props.onPressItem(index);
             else if (typedChild.props.onPress)
             typedChild.props.onPress(index);
          },
          toggled: this.initialToggleIndex === index || this.state.activeButtonIndex === index,
        });
        return <View style={styles.buttonContainer}><Composition source="Auryn_Btn-Nav-List-Container">{button}</Composition></View>;
      };
    });

    return <ScrollRef scrollEnabled={false} name={this.props.name} horizontal={true}>
        {data}
      </ScrollRef>;
  }
}

const styles = StyleSheet.create({
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
