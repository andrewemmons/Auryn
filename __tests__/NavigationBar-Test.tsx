/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { NavigationBar, ToggleButton } from '../components';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<NavigationBar name="Nav-List"
  scrollEnabled={false}
  horizontal={true}
  focusable={true}
  onPressItem={jest.fn()}
  initialToggleIndex={0}>
      <ToggleButton/>
      <ToggleButton/>
      <ToggleButton/>
    </NavigationBar>).toJSON();
  expect(tree).toMatchSnapshot();
});
