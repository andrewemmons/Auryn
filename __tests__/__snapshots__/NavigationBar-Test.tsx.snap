// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ListRef
  DEPRECATED_sendUpdatedChildFrames={false}
  alwaysBounceHorizontal={true}
  alwaysBounceVertical={false}
  focusable={true}
  horizontal={true}
  initialToggleIndex={0}
  name="Nav-List"
  onContentSizeChange={null}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onPressItem={[MockFunction]}
  onResponderGrant={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminationRequest={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onScrollShouldSetResponder={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  scrollBarThumbImage={null}
  scrollEnabled={false}
  sendMomentumEvents={false}
  style={
    Array [
      Object {
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      },
      undefined,
    ]
  }
>
  <ScrollContentData
    collapsable={false}
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "marginHorizontal": 4,
          }
        }
      >
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Btn-Nav-List-Container"
        >
          <YiPushButtonView
            focusable={true}
            name="Btn-Nav-List"
            onCompositionDidLoad={[Function]}
            onFocus={[Function]}
            onPress={[Function]}
            onTimelinesLoaded={[Function]}
          >
            <TimelineRef
              autoplay={true}
              direction="forward"
              loop={false}
              name="Toggle-On"
              onAborted={[Function]}
              onCompleted={[Function]}
              onCompletedForward={[Function]}
              onCompletedReverse={[Function]}
              onCompositionDidLoad={[Function]}
              onPaused={[Function]}
              onPausedForward={[Function]}
              onPausedReverse={[Function]}
              onPlay={[Function]}
              onPlayForward={[Function]}
              onPlayReverse={[Function]}
              onStarted={[Function]}
              onStartedForward={[Function]}
              onStartedReverse={[Function]}
            />
          </YiPushButtonView>
        </Composition>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 4,
          }
        }
      >
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Btn-Nav-List-Container"
        >
          <YiPushButtonView
            focusable={true}
            name="Btn-Nav-List"
            onCompositionDidLoad={[Function]}
            onFocus={[Function]}
            onPress={[Function]}
            onTimelinesLoaded={[Function]}
          >
            <TimelineRef
              autoplay={false}
              direction="reverse"
              loop={false}
              name="Toggle-On"
              onAborted={[Function]}
              onCompleted={[Function]}
              onCompletedForward={[Function]}
              onCompletedReverse={[Function]}
              onCompositionDidLoad={[Function]}
              onPaused={[Function]}
              onPausedForward={[Function]}
              onPausedReverse={[Function]}
              onPlay={[Function]}
              onPlayForward={[Function]}
              onPlayReverse={[Function]}
              onStarted={[Function]}
              onStartedForward={[Function]}
              onStartedReverse={[Function]}
            />
          </YiPushButtonView>
        </Composition>
      </View>
      <View
        style={
          Object {
            "marginHorizontal": 4,
          }
        }
      >
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Btn-Nav-List-Container"
        >
          <YiPushButtonView
            focusable={true}
            name="Btn-Nav-List"
            onCompositionDidLoad={[Function]}
            onFocus={[Function]}
            onPress={[Function]}
            onTimelinesLoaded={[Function]}
          >
            <TimelineRef
              autoplay={false}
              direction="reverse"
              loop={false}
              name="Toggle-On"
              onAborted={[Function]}
              onCompleted={[Function]}
              onCompletedForward={[Function]}
              onCompletedReverse={[Function]}
              onCompositionDidLoad={[Function]}
              onPaused={[Function]}
              onPausedForward={[Function]}
              onPausedReverse={[Function]}
              onPlay={[Function]}
              onPlayForward={[Function]}
              onPlayReverse={[Function]}
              onStarted={[Function]}
              onStartedForward={[Function]}
              onStartedReverse={[Function]}
            />
          </YiPushButtonView>
        </Composition>
      </View>
    </View>
  </ScrollContentData>
</ListRef>
`;
